### PR TITLE
Only convert to angle if not alreadey

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Review by kosack — DISMISSED** [review]

**Review by kosack — DISMISSED** [review]

**Review by kosack — APPROVED** [review]

**Comment by kosack** [comment]
Thanks for figuring out the q.info thing! That's really subtle and surprising, but at least they seem to have some ideas to fix it in AstroPy

**Review by kosack — DISMISSED** [review]

**Review by kosack — APPROVED** [review]

File: ctapipe/instrument/camera/geometry.py [hunk 1]
```diff
@@ -800,7 +816,9 @@ def position_to_pix_index(self, x, y):
     @staticmethod
     def simtel_shape_to_type(pixel_shape):
         try:
-            return SIMTEL_PIXEL_SHAPES[pixel_shape]
+            shape, rotation = SIMTEL_PIXEL_SHAPES[pixel_shape]
+            # make sure we don't introduce a mutable global state
+            return shape, rotation.copy()
```
Comment (LukasNickel) [inline ctapipe/instrument/camera/geometry.py:821]: Why do we need the `.copy()`?

File: ctapipe/instrument/camera/geometry.py [hunk 2]
```diff
@@ -623,12 +636,15 @@ def rotate(self, angle):
             rotation angle with unit (e.g. 12 * u.deg), or "12d"
 
         """
+        angle = Angle(angle)
```
Comment (LukasNickel) [inline ctapipe/instrument/camera/geometry.py:639]: Maybe also allow an `Angle` here and avoid the conversion in that case?

File: ctapipe/instrument/camera/geometry.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe/instrument/camera/geometry.py:821]: angles are mutable and this injected a global, mutable variable into classes that should not have any reliance on global state.

File: ctapipe/instrument/camera/geometry.py (hunk 2 above)
Comment (maxnoe) [reply ctapipe/instrument/camera/geometry.py:639]: Rotate is not called often (not e.g. in every event)

**Review by kosack — APPROVED** [review]

**Review by HealthyPear — APPROVED** [review]

**Comment by maxnoe** [comment]
On second thought, I think a better / safer way would be to add a `copy=True` keyword to the `__init__` method and use `copy=False` in `__getitem__`.

What do you think?

**Pull request merged**